sponsorship_compassion/models/contracts.py

```diff
@@ -544,7 +544,7 @@ def write(self, vals):
             self.mapped("partner_id").update_number_sponsorships()
             old_partners.update_number_sponsorships()
 
-        if "correspondent_id" in vals:
+        if "correspondent_id" in vals and self.state in ["active"]:
             for record in self:
```

Comment:
`self.state` wouldn't work when the recordset has more than one sponsorship.

```suggestion
        if "correspondent_id" in vals:
            updated_correspondents = self.filtered("global_id")
            for record in updated_correspondents:
```

Reply:
Sorry didn't saw your suggestion